plugins/org.eclipse.fordiac.ide.debug.ui/src/org/eclipse/fordiac/ide/debug/ui/fb/FBLaunchConfigurationTab.java

```diff
@@ -98,12 +111,93 @@ protected Composite createEventComponent(final Composite parent) {
 		return group;
 	}
 
+	protected Composite createDebugTimeComponent(final Composite parent) {
+		Label debugTimeLabel;
+		final Group group = new Group(parent, SWT.BORDER);
+		GridLayoutFactory.swtDefaults().applyTo(group);
+		group.setText("Debug Time"); //$NON-NLS-1$
+
+		final Composite comp = new Composite(group, SWT.NONE);
+		GridLayoutFactory.swtDefaults().numColumns(3).applyTo(comp);
+		GridDataFactory.fillDefaults().grab(true, false).applyTo(comp);
+
+		systemTimeRadio = new Button(comp, SWT.RADIO);
+		systemTimeRadio.setText("System"); //$NON-NLS-1$
+		systemTimeRadio.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(final SelectionEvent e) {
+				updateLaunchConfigurationDialog();
+				setCheckbox(systemTimeRadio.getText());
+			}
+		});
+		GridDataFactory.fillDefaults().applyTo(systemTimeRadio);
+		incrementTimeRadio = new Button(comp, SWT.RADIO);
+		incrementTimeRadio.setText("Increment"); //$NON-NLS-1$
+		incrementTimeRadio.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(final SelectionEvent e) {
+				updateLaunchConfigurationDialog();
+				setCheckbox(incrementTimeRadio.getText());
+			}
+		});
+		GridDataFactory.fillDefaults().applyTo(incrementTimeRadio);
+		manualTimeRadio = new Button(comp, SWT.RADIO);
+		manualTimeRadio.setText("Manual"); //$NON-NLS-1$
+		manualTimeRadio.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(final SelectionEvent e) {
+				updateLaunchConfigurationDialog();
+				setCheckbox(manualTimeRadio.getText());
+			}
+		});
+		GridDataFactory.fillDefaults().applyTo(manualTimeRadio);
```

Comment:
I would suggest aligning these buttons in a column instead of in one row and adding a more descriptive label to explain what each of these modes mean, such as:
- Use system clock
- Use fixed clock with specified time
- Increment clock after each event by specified amount

Reply:
like that suggestion. also extracted all the strings and removed the wrong non-nls tags